app/jobs/backorder_job.rb

```diff
@@ -36,23 +36,27 @@ def self.place_backorder(order, linked_variants)
     orderer = FdcBackorderer.new(user)
     backorder = orderer.find_or_build_order(order)
     broker = load_broker(order.distributor.owner)
+    ordered_quantities = {}
 
     linked_variants.each do |variant|
       needed_quantity = -1 * variant.on_hand
-      offer = broker.best_offer(variant.semantic_links[0].semantic_id)
+      solution = broker.best_offer(variant.semantic_links[0].semantic_id)
 
-      line = orderer.find_or_build_order_line(backorder, offer)
-      line.quantity = line.quantity.to_i + needed_quantity
+      wholesale_quantity = (needed_quantity.to_f / solution.factor).ceil
+      retail_quantity = wholesale_quantity * solution.factor
```

Comment:
A comment to explain the calculation would be useful, took me a minute to follow what's happening. 
`factor` here is the number of items available in the retail variant, ie: as per your spec, there are 12 cans in 1 retail variant (a case in our example). 
Maybe we need to rename `factor` to something more descriptive or keep it as `contained_quantity` like in `FdcOfferBroker`